File: scripts/qa/sample-all.mjs
// Dump ALL cases in the DB (not a random sample) for full manual audit.
// Same format as sample-cases.mjs so apply-audit.mjs can reuse the parser.
import { createClient } from '@supabase/supabase-js';
import { writeFile, mkdir } from 'fs/promises';
import path from 'path';

const supa = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL, process.env.SUPABASE_SERVICE_ROLE_KEY);

const { data: cases, error } = await supa
  .from('cases')
  .select('id,title,industry,case_type,difficulty,problem_statement,interviewer_notes,ideal_structure,casebook_id,provenance')
  .order('casebook_id', { ascending: true })
  .order('title', { ascending: true });

if (error) { console.error(error); process.exit(1); }

console.log(`Total cases in DB: ${cases.length}`);

const lines = [];
lines.push('# CasePad — FULL Case Audit\n');
lines.push(`All ${cases.length} cases. Mark each:\n`);
lines.push('- `[x]` good — keep as-is');
lines.push('- `[~]` usable but rough — keep, can polish later');
lines.push('- `[ ]` bad — DELETE');
lines.push('\nThen run `node --env-file=.env.local scripts/qa/apply-audit.mjs --apply` (script reads case-audit.md).\n');
lines.push('---\n');

for (let i = 0; i < cases.length; i++) {
  const c = cases[i];
  const book = c.provenance?.casebook_title?.replace(/^manual__/, '').replace(/_20/g, ' ').replace(/\.pdf$/, '') || 'unknown';
  lines.push(`## ${i + 1}. \`[ ]\` ${c.title}`);
  lines.push(`**id:** \`${c.id}\` · **book:** ${book} · **industry:** ${c.industry} · **type:** ${c.case_type} · **difficulty:** ${c.difficulty}\n`);
  lines.push('### Problem statement');
  lines.push((c.problem_statement || '_(empty)_').slice(0, 1500));
  lines.push('');
  lines.push('### Interviewer notes (reveal-on-question)');
  if (Array.isArray(c.interviewer_notes) && c.interviewer_notes.length) {
    for (const n of c.interviewer_notes.slice(0, 5)) {
      const trig = (n.trigger_keywords || []).join(', ');
      const rev = (n.reveal_text || '').slice(0, 300);
      lines.push(`- **trigger:** ${trig.slice(0, 200) || '_(none)_'}`);
      lines.push(`  **reveal:** ${rev}`);
    }
    if (c.interviewer_notes.length > 5) lines.push(`- _(+${c.interviewer_notes.length - 5} more)_`);
  } else {
    lines.push('_(none)_');
  }
  lines.push('');
  lines.push('### Ideal structure');
  const s = c.ideal_structure || {};
  lines.push(`framework: ${s.framework ?? '_(none)_'}`);
  if (Array.isArray(s.branches) && s.branches.length) {
    for (const b of s.branches.slice(0, 6)) {
      const subs = (b.subnodes || []).map((x) => typeof x === 'string' ? x : JSON.stringify(x));
      lines.push(`- **${b.node}** → ${subs.join(', ').slice(0, 200) || '_(no subnodes)_'}`);
    }
  }
  if (Array.isArray(s.key_insights) && s.key_insights.length) {
    lines.push(`**insights:** ${s.key_insights.slice(0, 3).join(' · ').slice(0, 300)}`);
  }
  lines.push('\n---\n');
}

const out = path.resolve('docs/qa/case-audit.md');
await mkdir(path.dirname(out), { recursive: true });
await writeFile(out, lines.join('\n'));
console.log(`Wrote ${out}`);
process.exit(0);

File: scripts/qa/sample-all.py
"""Dump all cases from Supabase to docs/qa/case-audit.md.
Uses urllib + REST API to avoid the Node.js + Supabase + Windows libuv crash."""
import json, os, urllib.request, urllib.parse, sys, pathlib

URL = os.environ.get('NEXT_PUBLIC_SUPABASE_URL')
KEY = os.environ.get('SUPABASE_SERVICE_ROLE_KEY')
if not URL or not KEY:
    print('Missing env vars. Source from .env.local first.', file=sys.stderr); sys.exit(1)

req = urllib.request.Request(
    f'{URL}/rest/v1/cases?select=id,title,industry,case_type,difficulty,problem_statement,interviewer_notes,ideal_structure,casebook_id,provenance&order=casebook_id.asc,title.asc',
    headers={'apikey': KEY, 'Authorization': f'Bearer {KEY}'},
)
with urllib.request.urlopen(req) as r:
    cases = json.loads(r.read())

print(f'Total cases in DB: {len(cases)}')

lines = ['# CasePad — FULL Case Audit\n']
lines.append(f'All {len(cases)} cases. Mark each:\n')
lines.append('- `[x]` good — keep as-is')
lines.append('- `[~]` usable but rough — keep, can polish later')
lines.append('- `[ ]` bad — DELETE\n')
lines.append('Then run `node --env-file=.env.local scripts/qa/apply-audit.mjs --apply`.\n')
lines.append('---\n')

def s(x):
    if isinstance(x, str): return x
    return json.dumps(x, ensure_ascii=False)

for i, c in enumerate(cases, 1):
    book = (c.get('provenance') or {}).get('casebook_title', 'unknown')
    book = book.replace('manual__', '').replace('_20', ' ').replace('.pdf', '')
    lines.append(f"## {i}. `[ ]` {c['title']}")
    lines.append(f"**id:** `{c['id']}` · **book:** {book} · **industry:** {c.get('industry','')} · **type:** {c.get('case_type','')} · **difficulty:** {c.get('difficulty','')}\n")
    lines.append('### Problem statement')
    lines.append((c.get('problem_statement') or '_(empty)_')[:1500])
    lines.append('')
    lines.append('### Interviewer notes (reveal-on-question)')
    notes = c.get('interviewer_notes') or []
    if notes:
        for n in notes[:5]:
            trig = ', '.join(s(t) for t in (n.get('trigger_keywords') or []))[:200]
            rev = (n.get('reveal_text') or '')[:300]
            lines.append(f"- **trigger:** {trig or '_(none)_'}")
            lines.append(f"  **reveal:** {rev}")
        if len(notes) > 5:
            lines.append(f'- _(+{len(notes)-5} more)_')
    else:
        lines.append('_(none)_')
    lines.append('')
    lines.append('### Ideal structure')
    struct = c.get('ideal_structure') or {}
    lines.append(f"framework: {struct.get('framework') or '_(none)_'}")
    branches = struct.get('branches') or []
    for b in branches[:6]:
        subs = ', '.join(s(x) for x in (b.get('subnodes') or []))[:200]
        lines.append(f"- **{b.get('node','?')}** → {subs or '_(no subnodes)_'}")
    insights = struct.get('key_insights') or []
    if insights:
        ins = ' · '.join(s(x) for x in insights[:3])[:300]
        lines.append(f"**insights:** {ins}")
    lines.append('\n---\n')

out = pathlib.Path('docs/qa/case-audit.md')
out.parent.mkdir(parents=True, exist_ok=True)
out.write_text('\n'.join(lines), encoding='utf-8')
print(f'Wrote {out} ({len(cases)} cases)')
